### PR TITLE
Improve development workflow

### PR DESCRIPTION
**Pull request by jeevb**
# TL;DR
Adds quality of life improvements for developers/contributors

## Complete description
This PR adds the following:
- Makefile with recipes to execute common development workflows
- Code auto-formatting with black/isort
- Linting with flake8
- Requirements management with pip-tools and freezes versions for `flytekit` and dev requirements

Code formatters were run and all flake8 errors were fixed as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,23 +81,44 @@ Or install them with the `all` directive. `all` defaults to Spark 2.4.x currentl
...
+
+```bash
+source ~/.virtualenvs/flytekit/bin/activate
+make fmt
 ```
 
-#### Execute
+### Testing
+
+Flytekit is Python 2.7+ compatible, so when feasible, it is recommended to test with both Python 2 and 3.
```
Comment (wild-endeavor) [inline README.md:117]: Let's kill this line... I've been actively removing Python 2.7 stuff.  I think the new version we're targeting will only be Python 3.6+.

**Comment by wild-endeavor** [comment]
Thank you thank you thank you thank you.  🙏   This [commit, a12dab832b1bbf16b997932a634cd815102d0da1](https://github.com/lyft/flytekit/pull/154/commits/a12dab832b1bbf16b997932a634cd815102d0da1) was the only mechanical code change right?

**Comment by jeevb** [comment]
The flake8 fixes are here:
https://github.com/lyft/flytekit/pull/154/commits/45d8bac88dabda15ab15e61a31eab7cd5b2b110f
https://github.com/lyft/flytekit/pull/154/commits/380ec4508fcedc872f42f166cf1008cac783391f
https://github.com/lyft/flytekit/pull/154/commits/a12dab832b1bbf16b997932a634cd815102d0da1

These should be fairly easy to peruse.

File: setup.py [hunk 2]
```diff
@@ -47,7 +47,7 @@
         "six>=1.9.0,<2.0.0",
         "sortedcontainers>=1.5.9<3.0.0",
         "statsd>=3.0.0,<4.0.0",
-        "urllib3>=1.22,<1.26",
+        "urllib3>=1.22,<2.0.0",
```
Comment (jeevb) [inline setup.py:50]: Updated to match https://github.com/lyft/flytekit/pull/155 to avoid merge conflict.

**Pull request merged**